pkg/operator/certrotation/signer.go

```diff
@@ -104,13 +102,18 @@ func (c RotatedSigningCASecret) EnsureSigningCertKeyPair(ctx context.Context) (*
 
 		LabelAsManagedSecret(signingCertKeyPairSecret, CertificateTypeSigner)
 
+		modified = true
+		signerUpdated = true
+	}
+
+	if modified {
 		actualSigningCertKeyPairSecret, _, err := applyFn(ctx, c.Client, c.EventRecorder, signingCertKeyPairSecret)
 		if err != nil {
 			return nil, false, err
 		}
 		signingCertKeyPairSecret = actualSigningCertKeyPairSecret
-		signerUpdated = true
```

Comment:
Could we remove `signerUpdated` now that we have `modified`?

Reply:
We still need it in `return` - `modified` covers any change to the secret, but `signerUpdate` is a flag for signer contents change (which triggers CA bundle update in `syncWorker`)